build.gradle.kts

```
@@ -177,4 +177,8 @@ gradle.projectsEvaluated {
                 }
             }
     }
+
+    tasks.named("build") {
```

@romtsn is there a nicer way of declaring the dependency?
Like `build.dependsOn("aggregateJavadocs")`?

---

@romtsn is the Gradle expert, but I guess `dependsOn` does the job.

So `make all` (called in CI) calls `make dryRelease`, maybe we make the `aggregateJavadocs` as part of the Dry release instead?

---

@marandaneto we'd only have to change the task name like this:

```suggestion
    tasks.named("publishToMavenLocal") {
```

Then it would only run on `make dryRelease` not on `./gradlew build`

---

but why do we need to publish it to mavenLocal and not mavenCentral? Just to make the CI pass?

---

Because the makefile says:

```
# do a dry release (like a local deploy)
dryRelease:
	./gradlew publishToMavenLocal --no-daemon --no-parallel
```

---

@romtsn no, it's a way of testing our release generation pipeline.
`publishToMavenLocal` triggers every task needed in order to generate jars, aars, poms, etc.
Uploading to maven central is done thru `craft`.

---

@adinauer maybe changing `dryRelease` only would do:

`./gradlew aggregateJavadocs publishToMavenLocal --no-daemon --no-parallel`

---

I've updated the PR to only change the `Makefile`

---

@adinauer if it works, LGTM.

---

ah ok so only the dryRelease was the problem then it makes sense

---

@romtsn yes, release failed due to Javadoc generation failing. Now we can run the check on every PR to avoid the late detection.

`make all` ran successfully here https://github.com/getsentry/sentry-java/runs/6963622852?check_suite_focus=true

I'll push a problematic commit to test if it actually fails.

---

Fails this PR on problematic javadoc comment, see here https://github.com/getsentry/sentry-java/runs/6963892775?check_suite_focus=true#step:5:7128